### PR TITLE
10/8/24 PR added 4 new Pids

### PR DESCRIPTION
**Pull request by Kilobravo18**
Evap: (Percent) (Range 0 to 100 *see below)
VSS: https://github.com/DIMO-Network/model-garage/blob/main/pkg/schema/spec/vss_rel_4.2-DIMO-b2679e5c.csv#L1206
Elastic: https://kibana.team.dimo.zone/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(breakdownField:data.evap,columns:!(data.evap,data.model),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,field:source,index:'325db2b7-a050-4589-b346-2bfd66929719',key:source,negate:!f,params:(query:dimo%2Fintegration%2F27qftVRWQYpVDcO5DltO5Ojbjxk),type:phrase),query:(match_phrase:(source:dimo%2Fintegration%2F27qftVRWQYpVDcO5DltO5Ojbjxk)))),index:'325db2b7-a050-4589-b346-2bfd66929719',interval:auto,query:(language:kuery,query:'data.evap%20:%20*%20'),sort:!(!(time,desc)))
Notes:

0% to 5%: When the engine is at idle or just started, the purge valve may be mostly closed, allowing little to no fuel vapor to be sent to the engine.
5% to 15%: During light throttle conditions, the purge valve opens slightly to allow some vapor flow into the engine.
15% to 45%: Under moderate throttle, the purge valve opens more to manage increased vapor flow.
45% to 75%: During higher load conditions (such as acceleration), the purge valve opens significantly to send more vapor to the engine.
75% to 100%: During very high load conditions or certain diagnostic cycles, the purge valve may open fully.
atfTemperature: (C) (70ish to 120ish *see below)
VSS: https://github.com/DIMO-Network/model-garage/blob/main/pkg/schema/spec/vss_rel_4.2-DIMO-b2679e5c.csv#L144
Elastic: https://kibana.team.dimo.zone/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(breakdownField:data.atfTemperature,columns:!(data.model,data.atfTemperature,data.make),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,field:source,index:'325db2b7-a050-4589-b346-2bfd66929719',key:source,negate:!f,params:(query:dimo%2Fintegration%2F27qftVRWQYpVDcO5DltO5Ojbjxk),type:phrase),query:(match_phrase:(source:dimo%2Fintegration%2F27qftVRWQYpVDcO5DltO5Ojbjxk)))),index:'325db2b7-a050-4589-b346-2bfd66929719',interval:auto,query:(language:kuery,query:'data.atfTemperature%20:%20*'),sort:!(!(time,desc)))
Notes:

Below 70°C (158°F): The transmission is relatively cool, often seen when the car has just started or is driving in cold conditions.
70°C to 100°C (158°F to 212°F): This is the normal operating range, where the transmission fluid is working efficiently, and transmission components are protected.
100°C to 120°C (212°F to 248°F): The fluid is starting to get hot, which can lead to accelerated wear of transmission components if sustained over time.
Above 120°C (248°F): The transmission fluid is too hot, and overheating can occur, potentially leading to fluid breakdown and damage to transmission components.
oxygenSensor1 (v) (0-5)
VSS:https://github.com/DIMO-Network/model-garage/blob/main/pkg/schema/spec/vss_rel_4.2-DIMO-b2679e5c.csv#L1174
Elastic: https://kibana.team.dimo.zone/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(columns:!(data.make,data.model,data.oxygenSensor1),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,field:source,index:'325db2b7-a050-4589-b346-2bfd66929719',key:source,negate:!f,params:(query:dimo%2Fintegration%2F27qftVRWQYpVDcO5DltO5Ojbjxk),type:phrase),query:(match_phrase:(source:dimo%2Fintegration%2F27qftVRWQYpVDcO5DltO5Ojbjxk)))),index:'325db2b7-a050-4589-b346-2bfd66929719',interval:auto,query:(language:kuery,query:'data.oxygenSensor1%20:%20*'),sort:!(!(time,desc)))

oxygenSensor2 (v) (0-5)
VSS: https://github.com/DIMO-Network/model-garage/blob/main/pkg/schema/spec/vss_rel_4.2-DIMO-b2679e5c.csv#L1178
Elastic: https://kibana.team.dimo.zone/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(columns:!(data.make,data.model,data.oxygenSensor2),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,field:source,index:'325db2b7-a050-4589-b346-2bfd66929719',key:source,negate:!f,params:(query:dimo%2Fintegration%2F27qftVRWQYpVDcO5DltO5Ojbjxk),type:phrase),query:(match_phrase:(source:dimo%2Fintegration%2F27qftVRWQYpVDcO5DltO5Ojbjxk)))),index:'325db2b7-a050-4589-b346-2bfd66929719',interval:auto,query:(language:kuery,query:'data.oxygenSensor2%20:%20*'),sort:!(!(time,desc)))

For Oxygen Sensor this is a wideband oxygen pid that is why the range is 0-5. These sensors switch between high and low.
generally speaking the ranges should be between 0.3 to 0.7 because:

Lean mixture (excess oxygen): Voltage will be low, around 0.1 to 0.3 volts.
Rich mixture (excess fuel): Voltage will be higher, around 0.7 to 0.9 volts.
Other Notes:

For AtfTemperature please ignore Silverado 3500 HD, Silverado 2500 HD and Opel Astra...I know these are out of range with wild readings, however they are getting good data back on the rest of the template. Maybe at a later date we make a new template for those MMYs, but right now this is the best spot for them.
For Oxygen Sensors 1 &2 please ignore Chevy Volt readings again same problem as above. It is a plug in hybrid.

### CODE REVIEW

File: pkg/schema/spec/definitions.yaml [hunk 1]
```diff
@@ -372,3 +372,52 @@
...
+      isArray: false
+  requiredPrivileges:
+    - VEHICLE_NON_LOCATION_DATA 
+- vspecName: Vehicle.OBD.FuelPressure
+  conversions: 
+    - originalName: fuelTankPressure
+      originalType: float64
+      isArray: false
+  requiredPrivileges:
+    - VEHICLE_NON_LOCATION_DATA
```
Comment (KevinJoiner) [inline pkg/schema/spec/definitions.yaml:395]: Is this new?

File: pkg/schema/spec/definitions.yaml [hunk 2]
```diff
@@ -372,3 +372,52 @@
...
+- vspecName: Vehicle.OBD.FuelPressure
+  conversions: 
+    - originalName: fuelTankPressure
+      originalType: float64
+      isArray: false
+  requiredPrivileges:
+    - VEHICLE_NON_LOCATION_DATA 
+- vspecName: evap
+  conversions: 
+    - originalName: Vehicle.OBD.CommandedEVAP
```
Comment (KevinJoiner) [inline pkg/schema/spec/definitions.yaml:398]: These names should be swapped

**Review by KevinJoiner — COMMENTED** [review]
Your definitions will need to be fixed so the generation can run.
I appreciate the detailed description.

**Comment by KevinJoiner** [comment]
@elffjs What are your thoughts on adding AtfTemperature even though we have a couple of MMYs with values in the thousands?
Would you want to 
1. wait for new template for those MMY
2. Add something here that ignores values over a certain temp
3. just let it fly

File: pkg/schema/spec/definitions.yaml (hunk 1 above)
Comment (Kilobravo18) [reply pkg/schema/spec/definitions.yaml:395]: no those are from the last PR I did back 10/1.

File: pkg/schema/spec/definitions.yaml (hunk 2 above)
Comment (Kilobravo18) [reply pkg/schema/spec/definitions.yaml:398]: I dyslexia love....lol, corrected this.

**Comment by elffjs** [comment]
I think we just let it ride. We'll have a bunch of bad data and people seem to accept that.

**Pull request closed without merging**